Tutorials/WinForms/Classic/bind-the-data-grid-to-large-data-source.md

```diff
@@ -0,0 +1,301 @@
+# Bind the Data Grid to a large data source
+
+[Step 3](/implement-crud-functionality-with-xpo-objects.md)
+&nbsp;&nbsp;&nbsp;
+[Back to TOC](../../)
+
+## Create the edit order Form
+* Select the **EditCustomerForm** item in the **Solution Explorer** window and press **Ctrl+C** to copy the Form.
+* Press **Ctrl+V** to create a copy of an EditCustomerForm and rename it to **EditOrderForm**.
+* Change the class and constructor name to **EditOrderForm**. Open the *CustomersListForm.Designer.cs* file and rename the class there. If this file is hidden, use the **Show All Files** toolbar item in the Solution Explorer window.
+* Open the `EditOrderForm` designer and change the `Text` property to **Edit Order**.
+* Rename the `CustomerBindingSource` component to **OrderBindingSource**.
+* Rebuild the project and set the `OrderBindingSource.ObjectClassInfo` property to **DxSample.DataAccess.Order**.
+* Set the `OrderBindingSource.DisplayableProperties` property to **ProductName;OrderDate;Freight;Customer!Key**. The last property name (**Customer!Key** is a [virtual property](https://docs.devexpress.com/XPO/3113/concepts/property-descriptors) designed for LookUp editors; see also: [How to: Bind an XPCollection to a LookUp](https://docs.devexpress.com/XPO/2000/examples/how-to-bind-an-xpcollection-to-a-lookup)).
+* Rename the `CustomerLayoutControl` component to **OrderLayoutControl**.
+* Rebuild the project.
+* Select the `OrderLayoutControl` component on the design surface. 
+* Click the [smart-tag](https://docs.microsoft.com/en-us/dotnet/framework/winforms/controls/performing-common-tasks-using-smart-tags-on-wf-controls) glyph. If the [smart-tag](https://docs.microsoft.com/en-us/dotnet/framework/winforms/controls/performing-common-tasks-using-smart-tags-on-wf-controls) glyph is not visible, press **Esc** several times to navigate from the selected `LayoutGroup` to the `LayoutControl`.
+* Select the **Retrieve Fields** item to open the **Select Binding Source** wizard.
+* Click the **Next** button and choose these properties and editors on the **Manage Data Bindings** screen:
+    * ProductName, TextEdit
+    * OrderDate, DateEdit
+    * Freight, CalcEdit
+    * Customer, LookupEdit 
+* Click the **Finish** button, delete the **First Name** and **Last Name** layout items, reorder the rest layout items to create a layout similar to the `EditCustomerForm`, and save the changes.
+* Add the `XPBindingSource` component to the `Form` and rename it to **CustomersBindingSource**.
+* Rebuild the project.
+* Set the `CustomersBindingSource.ObjectClassInfo` property to **DxSample.DataAccess.Customer**.
+* Set the `CustomersBindingSource.DisplayableProperties` property to **Oid;ContactName**.
+* Select the `lookUpEdit1` editor placed near the **Customer** label and click the [smart-tag](https://docs.microsoft.com/en-us/dotnet/framework/winforms/controls/performing-common-tasks-using-smart-tags-on-wf-controls) glyph. 
```

Comment:
"placed near" is not very specific. Is it next to it?

Reply:
Yes, next to it. However, `lookUpEdit1` already identifies the editor. There is no need to specify the location. I removed this part.

de77111926f6f05420c54fd1ecf9125e31ef880d